thrift/src/test/java/com/linecorp/armeria/client/thrift/ThriftOverHttpClientTest.java

```diff
@@ -303,6 +305,50 @@ public void onError(Exception exception) {
         });
     }
 
+    @ParameterizedTest
+    @ArgumentsSource(ParametersProvider.class)
+    void contextCaptorSync(
+            ClientFactory clientFactory, SerializationFormat format, SessionProtocol protocol)
+            throws Exception {
+        withTimeout(() -> {
+            final HelloService.Iface client = Clients.newClient(clientFactory,
+                                                                uri(Handlers.HELLO, format, protocol),
+                                                                Handlers.HELLO.iface(), clientOptions);
+            Clients.captureNextContext();
```

Comment:
Just brainstorming, would using a capturor be easier to understand than two static methods? (it would look more like mockito which is common knowledge in Java so it could be a good mental model reference point for users).

```java
ContextCaptor captor = Clients.newContextCaptor();
client.hello();
ctx = captor.get();
```

We could also consider this pattern - I personally like it the best since it gives explicit scoping to capturing, but it could feel a bit obtuse.

```java
final ClientRequestContext ctx1;
final ClientRequestContext ctx2;

try (ContextCaptor captor = Clients.newContextCaptor()) {
  client.hello("kukuman");
  ctx1 = captor.get();
  client.hello("fufuman");
  ctx2 = captor.get();

  for (ClientRequestContext ctx : captor.getAll()) {
    System.out.println(ctx);
  }
}
```
```

Reply:
The second one was the initial prototype but it was way too verbose, so I ditched it. :sweat_smile: 

The first one looks nice. Let me give it a try.

Reply:
Done

Reply:
I realized if the client requests failed with an exception, there will be a context leaked to ThreadLocal. It's not huge since it gets reset next time but in a server with a lot of threads I guess it could be somewhat significant.

Are you sure with not using `SafeCloseable` to ensure the threadlocal is cleaned up?